### ISSUE TITLE
Ability to specify options (e.g. subdomains) for custom basemaps/overlays in configuration

### ISSUE THREAD

**Issue by littlehelper** [nrenner/brouter-web#51]
Hi,
Is is possible to specifiy options like subdomains (refer to http://leafletjs.com/reference.html#tilelayer) for custom basemaps overlays in the configuration?
Like...

`BR.conf.baseLayers = {'Map name': {url: 'http://{s}.exampleserver.com/?x={x}&y={y}&z={z}', options: {subdomains: '0123', maxZoom: 18} } };`

I think about using the following options:
- subdomains
- attribution
- minZoom
- maxZoom
- opacity
- tms, zoomOffset, zoomReverse, etc.

Greetings

**Comment by nrenner** [comment]
Currently, only the URL can be set, but I already suspected that this is too simple and one might want to specify some options. Ideally, tile providers would also provide some meta data, e.g. via [TileJSON](https://github.com/mapbox/tilejson-spec), but that seems to be only supported by Mapbox themselves.

As an alternative to your example I could imagine using a simple object that gets passed as `TileLayer` `options` with an additional `url` property as a convention:

`BR.conf.baseLayers = {'Map name': {url: 'http://{s}.exampleserver.com/?x={x}&y={y}&z={z}', subdomains: '0123', maxZoom: 18} };`

For now, you would need to add the layer code in [Map.js](https://github.com/nrenner/brouter-web/blob/master/js/Map.js) (current master) and [rebuild](https://github.com/nrenner/brouter-web#build) the project.

Do you have any specific layers in mind? I'm open to suggestions for additional layers to add if they're of general interest.

**Comment by bagage** [comment]
Similarly, this is now implemented @nrenner isn't it?

**Comment by nrenner** [comment]
Not quite, the configurations for the new layer tree (#146) require a build step and are more complex, so I'm not sure yet if that should be a replacement for the config.js custom layers. Currently both options exist.

**Comment by littlehelper** [comment]
Thanks to the new features of brouter.de/brouter-web I can add custom map layers without the need for setting up an own server any more - that is great. :)

Would it be possible to add the option to specify at least the subdomain format (e.g. '0123' vs 'abc') when adding a custom layer at brouter.de/brouter-web ?
